service/history/engine/engineimpl/register_domain_failover_callback.go

```diff
@@ -53,107 +53,121 @@ func (e *historyEngineImpl) registerDomainFailoverCallback() {
 	//		failover min / max task levels calculated & updated to shard (using shard lock) -> failover start
 	// above 2 guarantees that failover start is after persistence of the task.
 
-	failoverPredicate := func(shardNotificationVersion int64, nextDomain *cache.DomainCacheEntry, action func()) {
```

Comment:
this large function has been decomposed into a few methods. It's not super easy to read as a diff, but hopefully it's more readable with a degree of decomposition